src/template/componentsTypescript.ts

```diff
@@ -1,67 +1,117 @@
-export const felaComponent = (name: string, dependencies: boolean) => `${
+const getExtendProp = (
+  typescriptFelaTheme: boolean,
+  typescriptFelaExtendProp: boolean
+) =>
+  typescriptFelaExtendProp
+    ? `extend?: ${
+        typescriptFelaTheme
+          ? "RulesExtend<typeof felaRules>;"
+          : "Partial<Record<keyof typeof felaRules, TRule>>;"
+      }`
+    : "";
+
+const getExtendPropImport = (
+  typescriptFelaTheme: boolean,
+  typescriptFelaExtendProp: boolean
+) =>
+  typescriptFelaExtendProp
+    ? typescriptFelaTheme
+      ? `import type { RulesExtend } from 'styles/theme';`
+      : "import type { TRule } from 'fela';"
+    : "";
+
+export const felaComponent = (
+  name: string,
+  dependencies: boolean,
+  typescriptFelaTheme: boolean,
+  typescriptFelaExtendProp: boolean
+) => `${
   dependencies
-    ? "import { React, FC, FelaWithStylesProps } from '../../dependencies';"
-    : "import React, { FC } from 'react';\nimport type { FelaWithStylesProps } from 'react-fela';"
+    ? "import { React, FelaWithStylesProps } from '../../dependencies';"
+    : "import React from 'react';\nimport type { FelaWithStylesProps } from 'react-fela';"
```

Comment:
From React v. 17, there is no need to import React for JSX syntax. I suggest removing it in all templates since all new projects will have version > 17.

Reply:
👍

Reply:
I just realized it can happen that developers will use the new version on older projects with React 16 😅 . But in that case they can import React manually, what do you think? 🤔

Reply:
That's a wise point. Well, I think the best option is to add another option `includeReactImport`, and set it to false by default. Then make the release with a new major version and alert users they need to opt-in for it in order to maintain the old behavior.

Reply:
That's the right solution 👏